Get-DevConsole/InstallDevConsole.psm1

```diff
@@ -0,0 +1,379 @@
+            }
+        }
+    }   
+    catch
+    {
+        Write-Message -color Red -message "Unable to move files from $currentInstallationDirectory"
+        Write-Message -color Red -message "Error: $_"
+    }
+    
+}
+
+################################################################################
+# Clean up items older than 1 month from old cache directory
+################################################################################
+Function Clear-OldDirectory ($isFirstSetup)
+{
+    if($false -eq $isFirstSetup)
+    {
+        try
+        {
+            $oldCacheItems =  Get-ChildItem -Path "$oldCacheDirectory"
+    
+            foreach ($items in $oldCacheItems)
+            {
+                if ($items.LastWriteTime -lt (Get-Date).AddMonths(-1))
+                {
+                    Remove-Item -Path $items.FullName -Recurse -Force
+                }
+            }
+        }
+        catch
+        {
+            Write-Message -color Red -message "Unable to delete items older than 1 month from the directory $oldCacheDirectory"
+            Write-Message -color Red -message "Error: $_"
+        }
+    }
+  
+}
+
+################################################################################
+# Clear temporary download directory after installation completes
+################################################################################
+Function Clear-TemporaryDownloadDirectoryectory ($isInstallationComplete)
+{
+    try 
+    {
+        $tempCache =  Get-ChildItem -Path "$temporaryDownloadDirectory"
+        if ($isInstallationComplete -eq $True)
+        {
+            foreach ($items in $tempCache)
+            {
+                Remove-Item -Path $items.FullName -Force
+            }
+        } 
+    }
+    catch 
+    {
+        Write-Message -color Red -message "Unable to delete the ZIP file from the directory $temporaryDownloadDirectory"
+        Write-Message -color Red -message "Error: $_"
+    }
+}
+
+################################################################################
+# Updates the DevConsole on the local machine
+################################################################################
+Function Install-DevConsole
+{
+    # URL of the console tool.
+    $URL = "https://btdevconsole.azurewebsites.net"
+
+
+    # Install Directories
+    $devConsoleDirectory = "$env:USERPROFILE\Documents\Migration Service"
+    $oldCacheDirectory = "$devConsoleDirectory\Old"
+    $currentInstallationDirectory = "$devConsoleDirectory\Current"
+    $temporaryDownloadDirectory = "$devConsoleDirectory\Temp"
+
+    
+    # Start main process
+
+    # Main data gathering
+    $cookies = Get-Cookies -URL $URL
+
+    if($false -ne $cookies)
+    {
+        $devConsoleBuildFullFileName = Get-DevConsoleBuildFullFileName -URL $URL -cookies $cookies
+    }
+    
+    if($false -ne $devConsoleBuildFullFileName)
+    {
+        $isFirstSetup = Test-IfUpdateIsNeeded 

```

Comment:
You are declaring a variable within an "if" statement. If the condition is false, the variable "isFirstSetup" is never set. That is what happened to me, and I got the error, "The variable '$isFirstSetup' cannot be retrieved because it has not been set." **You should initialize the variable outside of the "if" first, and provide a default value.**

Reply:
The variable is declared on line 294.  Or I'm missing what you mean.

Reply:
I am referring to $isFirstSetup. If $devConsoleBuildFullFileName is false, the variable $isFirstSetup is never created, so the "if" statement on line 303 will give an error saying the variable $isFirstSetup does not exist.

Reply:
Fixed